zarr/tests/test_storage.py

```diff
@@ -1021,6 +1021,19 @@ def test_create(self):
         with pytest.raises(PermissionError):
             g.data[:] = 1
 
+    def test_modify_consolidated(self):
+        import zarr
+        url = "file://" + tempfile.mkdtemp()
+
+        # create
+        root = zarr.open_group(url, mode="w")
+        root.zeros('baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
+        zarr.consolidate_metadata(url)
+
+        # reopen and modify
+        root = zarr.open_consolidated(url, mode="r+")
+        root["baz"][0, 0] = 7
```

Comment:
Should verify that this did indeed change the data.
Also would add a case for each mode which attempts to change the metadata - should succeed for mode "w" and fail for "r+".

Reply:
`open_consolidated` may only be called with modes ("r" and "r+"). I've added test for the exeption on those and added a test on "w-" for `open_consolidated.

If you prefer, I can also add the "a", "w" and "w-" tests here, but they won't accept changes to the metadata, but will fail on opening (which is likely already covered by the tests in convenience).

Reply:
> I can also add the "a", "w" and "w-" tests here

Not necessary.

Maybe a test showing an exception when you open with "r" and try to write data? I know this was always an exception. If you think it's already covered elsewhere, I'm happy to take your word for it.

Reply:
+1